Closes #623 | Add/Update Dataloader MedEV

patrickamadeus opened the pull request:
Closes #623 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [.] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### Tests
<img width="1209" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/65403659/fbd4bdef-f274-4f87-9715-48fe4411c7b6">

elyanah-aco commented on seacrowd/sea_datasets/medev/medev.py:
```diff
@@ -0,0 +1,155 @@
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        with open(filepath["en"], "r") as f:
```
Let's set the encoding so this run on all systems - mine (Windows) doesn't run without it.
```suggestion
        with open(filepath["en"], "r", encoding="utf-8") as f:
```

elyanah-aco commented on seacrowd/sea_datasets/medev/medev.py:
```diff
@@ -0,0 +1,155 @@
+        """Yields examples as (key, example) tuples."""
+        with open(filepath["en"], "r") as f:
+            en_lines = f.readlines()
+        with open(filepath["vie"], "r") as f:
```
Can you determine what encoding your computer uses for this file? Both `utf-8` and `latin-1` don't give me the intended result, and other encodings don't work at all.

patrickamadeus replied:
Mine is `utf-8`. it's working well from my side.

<img width="1123" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/65403659/f40dad23-06ab-47df-9aba-cb8eb1b3f544">

elyanah-aco commented on seacrowd/sea_datasets/medev/medev.py:
```diff
@@ -0,0 +1,155 @@
+
+        if self.config.schema == "source":
+            for i, line in enumerate(en_lines + vie_lines):
+                yield i, {
+                    "text": line,
+                }
```
### 
```suggestion
                yield i, {
                    "id": str(i),
                    "text": line,
                }
```

elyanah-aco commented on seacrowd/sea_datasets/medev/medev.py:
```diff
@@ -0,0 +1,155 @@
+                    "id": str(i),
+                    "text_1": en_line,
+                    "text_2": vie_line,
+                    "text_1_name": "en",
```
We use ISO 639-3 codes for languages.
```suggestion
                    "text_1_name": "eng",
```

elyanah-aco reviewed (commented):
Thanks for the PR @patrickamadeus! Just suggesting some small changes:

elyanah-aco commented on seacrowd/sea_datasets/medev/medev.py:
```diff
@@ -0,0 +1,155 @@
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                }
```
```suggestion
                {
                    "id": datasets.Value("string")
                    "text": datasets.Value("string"),
                }
```

patrickamadeus replied:
Hi! I'll address @akhdanfadh 's suggestion after everything concluded from ur guys' side!

akhdanfadh reviewed (commented):
Thank you for the dataloader! Just add some minor edit and additional discussion.

patrickamadeus commented:
Hi @elyanah-aco ! I've addressed all of the suggestions! Appreciate the detailed review.

I will address suggestion from @akhdanfadh after your second opinion 🙏.

elyanah-aco reviewed (approved):
Thanks for the changes @patrickamadeus! OK now on my end (even the Vietnamese text)

patrickamadeus commented:
Hi all @akhdanfadh @elyanah-aco ! The minor language expand is done! Thank you for all of the reviews. 🙏

holylovenia commented:
Hi @akhdanfadh, I would like to let you know that we plan to finalize the calculation of the open contributions (e.g., dataloader implementations) **in 31 hours**, so it'd be great if we could wrap up the reviewing and merge this PR before then.

cc: @patrickamadeus

akhdanfadh reviewed (approved):
LGTM! Dataloader tested and worked. Merging in a bit.
